EstateManagement/Controllers/ContractController.cs

```diff
@@ -0,0 +1,220 @@
+            await this.Mediator.Send(command, cancellationToken);
+
+            // return the result
+            return this.Created($"{ContractController.ControllerRoute}/{contractId}/products/{productId}/transactionFees/{transactionFeeId}",
+                                new AddTransactionFeeForProductToContractResponse
+                                {
+                                    EstateId = estateId,
+                                    ContractId = contractId,
+                                    ProductId = productId,
+                                    TransactionFeeId = transactionFeeId
+                                });
+        }
+
+        /// <summary>
+        /// Creates the contract.
+        /// </summary>
+        /// <param name="estateId">The estate identifier.</param>
+        /// <param name="createContractRequest">The create contract request.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("")]
+        public async Task<IActionResult> CreateContract([FromRoute] Guid estateId,
+                                                        [FromBody] CreateContractRequestDTO createContractRequest,
+                                                        CancellationToken cancellationToken)
+        {
+            // Get the Estate Id claim from the user
+            Claim estateIdClaim = ClaimsHelper.GetUserClaim(this.User, "EstateId", estateId.ToString());
+
+            String estateRoleName = Environment.GetEnvironmentVariable("EstateRoleName");
+            if (ClaimsHelper.IsUserRolesValid(this.User, new[] {string.IsNullOrEmpty(estateRoleName) ? "Estate" : estateRoleName}) == false)
+            {
+                return this.Forbid();
+            }
+
+            if (ClaimsHelper.ValidateRouteParameter(estateId, estateIdClaim) == false)
+            {
+                return this.Forbid();
+            }
+
+            Guid contractId = Guid.NewGuid();
+
+            // Create the command
+            CreateContractRequest command = CreateContractRequest.Create(contractId, estateId, createContractRequest.OperatorId, createContractRequest.Description);
+
+            // Route the command
+            await this.Mediator.Send(command, cancellationToken);
+
+            // return the result
+            return this.Created($"{ContractController.ControllerRoute}/{contractId}",
+                                new CreateContractResponse
+                                {
+                                    EstateId = estateId,
+                                    OperatorId = command.OperatorId,
+                                    ContractId = contractId
+                                });
+        }
+
+        #endregion
+
+        #region Others
+
+        /// <summary>
+        /// The controller name
+        /// </summary>
+        public const String ControllerName = "contracts";
```

Comment:
Codacy found an issue: [Change this constant to a 'static' read-only property.](https://app.codacy.com/manual/stuart_ferguson1/EstateManagement/pullRequest?prid=5848159)